2020_RTSS_cpc_model_based_algorithm/src/main.rs

```diff
@@ -24,9 +27,21 @@ struct ArgParser {
 fn main() {
     let arg: ArgParser = ArgParser::parse();
     let mut dag = create_dag_from_yaml(&arg.dag_file_path);
-    let number_of_cores = arg.number_of_cores;
+    let homogeneous_processor = HomogeneousProcessor::new(arg.number_of_cores);
     prioritization_cpc_model::assign_priority_to_cpc_model(&mut dag);
-    let mut fixed_priority_scheduler =
-        FixedPriorityScheduler::new(&dag, &HomogeneousProcessor::new(number_of_cores));
-    let (_schedule_length, _) = fixed_priority_scheduler.schedule();
+    let mut fixed_priority_scheduler = FixedPriorityScheduler::new(&dag, &homogeneous_processor);
+
+    let (schedule_length, _) = fixed_priority_scheduler.schedule();
+    let now: DateTime<Utc> = Utc::now();
+    let date = now.format("%Y-%m-%d-%H-%M-%S").to_string();
+    let file_name = format!("{}-cpc-result", date);
+    let file_path = create_yaml_file(&arg.output_dir_path, &file_name);
+    dump_dag_set_info_to_yaml(&file_path, vec![dag.clone()]);
+    dump_processor_info_to_yaml(&file_path, homogeneous_processor);
+
+    let result = schedule_length < dag.get_head_period().unwrap();
```

Comment:
This algorithm assumes a constrained deadline, so this decision formula cannot be used.
It is preferable to receive a factor (<= 1.0) to multiply the period by in the command line argument and calculate the end-to-end deadline on that basis.
In addition, it would be great if the added facter were also output to the log.

Reply:
Get an end-to-end deadline as priority, or if none, use the period multiplied by the factor of arguments as the constrained deadline.

Also changed to output the factor of arguments to the log.